rclcpp_lifecycle/src/state.cpp

```diff
@@ -14,17 +14,38 @@
 
 #include "rclcpp_lifecycle/state.hpp"
 
-#include <lifecycle_msgs/msg/state.hpp>
+#include <string>
+#include <utility>
 
-#include <rcutils/allocator.h>
-#include <rcutils/strdup.h>
-#include <rcl_lifecycle/data_types.h>
+#include "lifecycle_msgs/msg/state.hpp"
 
-#include <string>
+#include "rcutils/allocator.h"
+#include "rcutils/strdup.h"
+
+#include "rcl_lifecycle/data_types.h"
 
 namespace rclcpp_lifecycle
 {
 
+void
+swap(rcl_lifecycle_state_t * lhs, const rcl_lifecycle_state_t * rhs)
+{
+  lhs->id = rhs->id;
+  lhs->label = rcutils_strndup(rhs->label, strlen(rhs->label), rcutils_get_default_allocator());
+}
```

Comment:
The semantic of this function is not [swap](http://en.cppreference.com/w/cpp/algorithm/swap). Therefore it should have a different name.

Same below.

Reply:
The name goes along with the copy-and-swap idiom.
https://stackoverflow.com/questions/3279543/what-is-the-copy-and-swap-idiom#3279550

Reply:
My point is that if I call `swap(a, b)` I expect those two instances to be "swapped" (a has the values form b, b has the values from a). The current code doesn't accomplish that: https://github.com/ros2/rclcpp/pull/395/files#diff-7731e343ff998d96d3dd4fcba8c97c69R45